tests/test_filtering.py

```diff
@@ -0,0 +1,251 @@
+            slice_axis=0,
+            plt=plt,
+        )
+
+    # All filters should be normalised aside from the exp filter
+    if filter_flag == 3:
+        # ratio of exponential and sphere volume integrals
+        norm_factor = (
+            2 * R_param**3
+            - R_param * np.exp(-R / R_param) * (2 * R_param**2 + 2 * R_param * R + R**2)
+        ) / (3 * R**3)
+    else:
+        norm_factor = 1
+    # firstly, make sure the filters are normalised
+    np.testing.assert_allclose(
+        input_box_centre.sum() * norm_factor, output_box_centre.sum(), rtol=1e-4
+    )
+
+    # then make sure we get the right shapes (more lenient for aliasing)
+    np.testing.assert_allclose(
+        output_box_centre, exp_output_centre, rtol=RELATIVE_TOLERANCE
+    )
+
+
+# since the filters are symmetric I'm doing an R vs value plot instead of imshowing slices
+def filter_plot(
+    inputs, outputs, binned_truths, truths, r_bins, r_grid, slice_index, slice_axis, plt
+):
+    if not (len(inputs) == len(binned_truths) == len(outputs)):
+        raise ValueError(
+            f"inputs {len(inputs)}, outputs {len(outputs)} and truths {len(binned_truths)}"
+            "must have the same length."
+        )
+
+    n_plots = len(inputs)
+    fig, axs = plt.subplots(
+        n_plots,
+        4,
+        figsize=(16.0, 12.0 * n_plots / 4.0),
+        layout="constrained",
+        squeeze=False,
+    )
+    fig.get_layout_engine().set(w_pad=2 / 72, h_pad=2 / 72, hspace=0.0, wspace=0.0)
+
+    r_cen = (r_bins[1:] + r_bins[:-1]) / 2
+
+    axs[0, 0].set_title("Input")
+    axs[0, 1].set_title("output")
+    axs[0, 2].set_title("Expected")
+    axs[0, 3].set_title("Radii")
+    for idx, (i, o, t, tt) in enumerate(zip(inputs, outputs, binned_truths, truths)):
+        axs[idx, 0].pcolormesh(
+            i.take(indices=slice_index, axis=slice_axis),
+            norm=Normalize(vmin=0, vmax=o.max()),
+        )
+        axs[idx, 1].pcolormesh(
+            o.take(indices=slice_index, axis=slice_axis),
+            norm=Normalize(vmin=0, vmax=o.max()),
+        )
+        axs[idx, 2].pcolormesh(
+            tt.take(indices=slice_index, axis=slice_axis),
+            norm=Normalize(vmin=0, vmax=o.max()),
+        )
+
+        stats_o = get_binned_stats(r_grid, o, r_bins, stats=["mean", "err1u", "err1l"])
+        axs[idx, 3].errorbar(
+            r_cen,
+            stats_o["mean"],
+            markerfacecolor="b",
+            elinewidth=1,
+            capsize=3,
+            markersize=5,
+            marker="o",
+            color="b",
+            yerr=[stats_o["err1l"], stats_o["err1u"]],
+            label="filtered grid",
+        )
+        axs[idx, 3].plot(r_cen, t, "m:", label="Expected")
+        axs[idx, 3].grid()
+        axs[idx, 3].set_xlabel("dist from centre")
+        axs[idx, 3].set_ylabel("cell value")
+
+        axst = axs[idx, 3].twinx()
+        axst.plot(r_cen, stats_o["mean"] / t - 1, "r-")
```

Comment:
Instead of dividing by the truth -- which could be very very close to zero, you might want to add some epsilon to the denominator here to account for machine-precision uncertainties (e.g. add 1e-16 to t so you don't get infs)